zrml/court/src/court_pallet_api.rs

```diff
@@ -1,12 +1,80 @@
-use frame_support::dispatch::{DispatchError, DispatchResultWithPostInfo, Weight};
-use zeitgeist_primitives::types::OutcomeReport;
+use crate::ResolutionCounters;
+use alloc::vec::Vec;
+use frame_support::dispatch::{DispatchError, DispatchResultWithPostInfo};
+use zeitgeist_primitives::types::{Market, MarketDispute, OutcomeReport};
 
 /// Court - Pallet Api
 pub trait CourtPalletApi {
+    type AccountId;
     type BlockNumber;
     type MarketId;
     type Origin;
 
+    // Market
+
+    /// Gets a market from the storage.
+    fn market(
+        market_id: &Self::MarketId,
+    ) -> Result<Market<Self::AccountId, Self::BlockNumber>, DispatchError>;
+
+    /// Inserts a market into the storage
+    fn insert_market(
+        market_id: &Self::MarketId,
+        market: Market<Self::AccountId, Self::BlockNumber>,
+    );
+
+    /// Mutates a given market storage
+    fn mutate_market<F>(market_id: &Self::MarketId, cb: F) -> Result<(), DispatchError>
+    where
+        F: FnOnce(&mut Market<Self::AccountId, Self::BlockNumber>);
+
+    /// Removes a market from the storage.
+    fn remove_market(market_id: &Self::MarketId) -> Result<(), DispatchError>;
```

Comment:
Both `court` and `PM` pallets use the `Markets` storage so it is necessary to create an common entry-point. In this case, `court` holds `Markets` and `PM` consumes the API

Reply:
Yes, this schema is semantically wrong, although that's what I could formulate given the constraints.
This subject needs more research and https://github.com/paritytech/substrate/issues/8320 might have some answers so I will open a related PR afterwards if it is not a problem.

Reply:
Yeah, I don't think this is the storage format we want in the final form, it makes Prediction Markets way too tightly coupled to Courts storage and would make replacing the Court pallet with something different in the future more difficult. The ideal interface between PM and Court should be that if we wanted to swap Courts with a <DISPUTE_MECHANISM>, it should be more or less a drop in replacement without impacting any markets in storage or possibly even requiring any migrations. The maximal version of this would be having multiple parallel dispute options at once, and they can be selected by the person making the dispute.

Reply:
Research of alternative methods to share storage would take more time so I simply created another pallet called `market-commons` that holds `Markets` and any other future common parameter between `PM` and `court`.